test/libdnf/rpm/test_package.cpp

```diff
@@ -0,0 +1,346 @@
+}
+
+
+void RpmPackageTest::test_equality() {
+    auto package1 = get_pkg("CQRlib-1.1.1-4.fc29.x86_64");
+    auto package2 = get_pkg("CQRlib-1.1.1-4.fc29.x86_64");
+    CPPUNIT_ASSERT(package1 == package2);
+
+    auto package3 = get_pkg("nodejs-1:5.12.1-1.fc29.x86_64");
+    CPPUNIT_ASSERT(package1 != package3);
+}
+
+
+void RpmPackageTest::test_get_id() {
+    get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_id();  // Can't really test for the value of the ID
+}
+
+
+void RpmPackageTest::test_get_name() {
+    CPPUNIT_ASSERT_EQUAL(std::string("CQRlib"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_name());
+}
+
+
+void RpmPackageTest::test_get_epoch() {
+    CPPUNIT_ASSERT_EQUAL(0lu, get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_epoch());
+    CPPUNIT_ASSERT_EQUAL(1lu, get_pkg("nodejs-1:5.12.1-1.fc29.x86_64").get_epoch());
+}
+
+
+void RpmPackageTest::test_get_version() {
+    CPPUNIT_ASSERT_EQUAL(std::string("1.1.1"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_version());
+}
+
+
+void RpmPackageTest::test_get_release() {
+    CPPUNIT_ASSERT_EQUAL(std::string("4.fc29"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_release());
+}
+
+
+void RpmPackageTest::test_get_arch() {
+    CPPUNIT_ASSERT_EQUAL(std::string("x86_64"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_arch());
+}
+
+
+void RpmPackageTest::test_get_evr() {
+    CPPUNIT_ASSERT_EQUAL(std::string("1.1.1-4.fc29"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_evr());
+    CPPUNIT_ASSERT_EQUAL(std::string("1:5.12.1-1.fc29"), get_pkg("nodejs-1:5.12.1-1.fc29.x86_64").get_evr());
+}
+
+
+void RpmPackageTest::test_get_nevra() {
+    CPPUNIT_ASSERT_EQUAL(std::string("CQRlib-1.1.1-4.fc29.x86_64"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_nevra());
+}
+
+
+void RpmPackageTest::test_get_full_nevra() {
+    CPPUNIT_ASSERT_EQUAL(
+        std::string("CQRlib-0:1.1.1-4.fc29.x86_64"),
+        get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_full_nevra()
+    );
+}
+
+
+void RpmPackageTest::test_get_group() {
+    CPPUNIT_ASSERT_EQUAL(
+        std::string("System Environment/Libraries"),
+        get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_group()
+    );
+}
+
+
+void RpmPackageTest::test_get_size() {
+    CPPUNIT_ASSERT_EQUAL(6324llu, get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_size());
+}
+
+
+void RpmPackageTest::test_get_download_size() {
+    CPPUNIT_ASSERT_EQUAL(6324llu, get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_download_size());
+}
+
+
+void RpmPackageTest::test_get_install_size() {
+    // TODO implement installing packages for tests
+    CPPUNIT_ASSERT_EQUAL(0llu, get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_install_size());
+}
+
+
+void RpmPackageTest::test_get_license() {
+    CPPUNIT_ASSERT_EQUAL(std::string("LGPLv2+"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_license());
+}
+
+
+void RpmPackageTest::test_get_sourcerpm() {
+    CPPUNIT_ASSERT_EQUAL(
+        std::string("CQRlib-1.1.1-4.fc29.src.rpm"),
+        get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_sourcerpm()
+    );
+    // TODO is it correct it's empty?
```

Comment:
I suggest that it is correct that it is empty. SRC rpm should have no src entry/. Also the data are originated from libsolv and metadata created by createrepo_c, therefore we have to keep it like it is. Please could you remove TODO note?